gateway: enable cli arguments for the IoT REST API Server

gvancuts opened the pull request:
* The IoT REST API Server Command-line arguments are enabled in Docker
  and transparently passed through. This allows to tweak its behaviour
  by changing the port and/or enabling https
* The README.md was updated to describe a couple of methods for deploying
  the Home Gateway function

Fix #147 

Signed-off-by: Geoffroy Van Cutsem <geoffroy.vancutsem@intel.com>

gvancuts commented:
@poussa @nagineni @uhofemei @ttzeng @ichbinblau @mjkadera please take a look

ichbinblau commented on gateway/Dockerfile:
```diff
@@ -31,6 +31,9 @@ WORKDIR /opt/SmartHome-Demo/gateway/
 RUN npm install
 
+# Install the certificates needed when using HTTPS
+RUN /bin/bash /opt/SmartHome-Demo/gateway/node_modules/iot-rest-api-server/config/generate-key-and-cert.sh
```
Hi, @gvancuts.  I have a minor finding. The script `generate-key-and-cert.sh` takes one parameter called FQDN. The FQDN is supposed to be the same as the service external facing host name (www.xxx.com etc) or simply the external IP address.  By default, it uses localhost which is 127.0.0.1 which is not right. As the Cloud portal does not validate the SSL certificate, it is not a problem for the time being.  If we are going to make the site secure in future, it will be a potential issue.

gvancuts replied:
Thanks @ichbinblau , this is good to know. We'll have to figure out a way of passing those certificates to a Docker container then (and remove that line) as we won't know what those are at build time. I'm assuming we will have to use the [data volume](https://docs.docker.com/engine/tutorials/dockervolumes/) mechanism for that (just like what you use for selectively starting OCF servers). It shouldn't be too hard, I'll take a look at it.

gvancuts commented:
New, enhanced version has been pushed. Let me know if you like the approach @ichbinblau (and all others). If so, I really ought to document it too before merging this ;)

ichbinblau commented on gateway/start-gateway-in-docker.sh:
```diff
@@ -11,17 +11,57 @@ export NODE_PATH=/opt/SmartHome-Demo/gateway/node_modules/
+			echo
+			echo "We will proceed using a certificate and private key that was generated for testing purposes only but be aware that the certificate is self-signed and browsers do not recognise it so you will get warnings."
+			echo
+			echo "We highly recommende that you get a proper certificate from a known certificate authority and its corresponding private key if you want to use 'https'"
```
A minor typo: recommende -> recommend.

ichbinblau commented:
@gvancuts, I like the enhancement, the warning on startup in particular. I have verified this PR with https enabled/disabled and both work.

gvancuts commented:
Nice catch @ichbinblau ! I have fixed the typo and have also now added some proper documentation in the README.md files to explain what this is and how to use it. I am planning to merge this next Tuesday if there is no concerns from anyone.

gvancuts commented:
I haven't heard any concerns or objections, merging this.
